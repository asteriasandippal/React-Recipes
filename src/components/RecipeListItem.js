import React from 'react';
import PropTypes from 'prop-types';

function RecipeListItem({
  recipe, onFavorited, favorites, onEventClick,
}) {
  return (
    <li
      className="py2 border-bottom border-bottom-dashed pointer"
      onClick={() => onEventClick(recipe.id)}
      onKeyDown={() => onEventClick(recipe.id)}
      role="presentation"
    >
      <span
        className="mr1"
        onClick={(e) => {
        e.stopPropagation();
        onFavorited(recipe.id);
      }}
        onKeyDown={() => onFavorited(recipe.id)}
        role="presentation"
        aria-label="favorite"
      >
        {favorites.includes(recipe.id) ? '✔' : '❑'}
      </span>
      <span>{recipe.name}</span>
      <span>{recipe.category}</span>
    </li>
  );
}

RecipeListItem.defaultProps = {
  recipe: {},
  onEventClick: () => {},
  onFavorited: () => {},
  favorites: [],
};

RecipeListItem.propTypes = {
  recipe: PropTypes.object,
  onEventClick: PropTypes.func,
  onFavorited: PropTypes.func,
  favorites: PropTypes.array,
};

export default RecipeListItem;

