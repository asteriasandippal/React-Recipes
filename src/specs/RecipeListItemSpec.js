import React from 'react';
import render from 'react-test-renderer';
import RecipeListItem from '../components/RecipeList';

const testRecipe = {
  id: 1,
  name: 'Test Recipe',
  category: 'Test Category',
};

describe('<RecipeListItem/>', () => {
  test('Should not break if no recipe passed', () => {
    const component = render.create(<RecipeListItem />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly render recipe', () => {
    const component = render.create(<RecipeListItem recipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('Should render favorited state', () => {
    const component = render.create(<RecipeListItem recipe={testRecipe} favorited />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
