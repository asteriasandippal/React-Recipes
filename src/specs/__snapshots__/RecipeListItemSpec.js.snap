// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem/> Should correctly render recipe 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeListItem/> Should not break if no recipe passed 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeListItem/> Should render favorited state 1`] = `
<ul
  className="list-reset"
/>
`;
